lib/wallaby/browser.ex

```diff
@@ -1220,11 +1221,23 @@ defmodule Wallaby.Browser do
   end
 
   defp path_for_screenshot(name) do
-    File.mkdir_p!(screenshot_dir())
     "#{screenshot_dir()}/#{name}.png"
   end
 
+  defp write_screenshot!(path, image_data) do
+    expanded_path = Path.expand(path)
+    :ok = expanded_path |> Path.dirname() |> File.mkdir_p!()
+
+    :ok = File.write!(expanded_path, image_data)
+
+    :ok
+  end
+
   defp screenshot_dir do
     Application.get_env(:wallaby, :screenshot_dir) || "#{File.cwd!()}/screenshots"
```

Comment:
This isn't related to your change, but this should probably be written as

```suggestion
    Application.get_env(:wallaby, :screenshot_dir, "#{File.cwd!()}/screenshots") 
```